### PR TITLE
Simplify godep licenses logic

### PR DESCRIPTION
**Pull request by thockin**
This code used to actually reach out to the internet to look for files.  This
is flaky, slow, and semantically WRONG.  The license that is upstream might
actually be different than what we have vendored.  Only look at local files.

This now passes back-to-back updates and verifies.

@quinton-hoole @mhrgoog @bgrant0607 - FYI
@zmerlynn @david-mcmahon - do you guys think this will satisfy legal?  The older form was WAY complicated and a regular source of pain.

### CODE REVIEW

File: hack/verify-godep-licenses.sh [hunk 1]
```diff
@@ -32,9 +32,9 @@ fi
 # to work with docker-machine on macs
 mkdir -p "${KUBE_ROOT}/_tmp"
 _tmpdir="$(mktemp -d "${KUBE_ROOT}/_tmp/kube-godep-licenses.XXXXXX")"
-echo "Created workspace: ${_tmpdir}"
+#echo "Created workspace: ${_tmpdir}"
```
Comment (quinton-hoole) [inline hack/verify-godep-licenses.sh]: nit: Remove dead comments here and two lines down?

**Comment by david-mcmahon** [comment]
@thockin We need to go get and cache the missing license data from the internet.  I don't know any other way around that.  

The old godep license process was several times slower than this and required hand-maintaining an exclusion list.  The artifact from that was simply a list of licenses and this new version gathers actual license content dynamically across all packages and does it several times faster.

I commented in the other issue.  What is the repro case here for the failure?

File: hack/verify-godep-licenses.sh (hunk 1 above)
Comment (thockin) [inline hack/verify-godep-licenses.sh]: I left them for the next poor debugger, but I could nix them.

**Comment by thockin** [comment]
@david-mcmahon Hard to pin down.  I spent hours trying to grok what was happening - back-to-back verify cycles would produce different results and then I caught it occasionally flaking in the (now removed) part.  I decided it was wrong and excised it.

**Comment by quinton-hoole** [comment]
@david-mcmahon FYI I ran it a few times on two different machines at different times, and it produced no output, or error message, ever.  I didn't debug it any further than that.

**Comment by quinton-hoole** [comment]
OK, I tested this, and it seems to do what's required.  Code looks fine to me. LGTM.

**Comment by quinton-hoole** [comment]
Fixes #26328

**Comment by quinton-hoole** [comment]
@thockin It seems that there are some merge conflicts on this?

**Comment by quinton-hoole** [comment]
@thockin It would be awesome if we could get this resolved ASAP, as there is a whole train of 1.3-blocking PR's queued up behind this one.

**Comment by a-robinson** [comment]
Manually merging to unblock dependent ubernetes PRs while submit queue is down. e2es passed, I assume this wouldn't have broken only unit tests.

**Comment by bgrant0607** [comment]
@quinton-hoole Writing "fixes #issue" in a PR comment doesn't close the issue. That needs to go in the PR description. Consequently, #26328 is still open.

**Comment by quinton-hoole** [comment]
@bgrant0607 notes, thanks!

**Comment by quinton-hoole** [comment]
@bgrant0607 FYI, according to [this](https://help.github.com/articles/closing-issues-via-commit-messages/) any commit message should do it, not only the PR description.  So I still don't actually understand why my commit message above didn't work.

**Comment by bgrant0607** [comment]
@quinton-hoole Github comments aren't commit messages.

**Pull request merged**